common/src/main/java/com/microsoft/identity/common/internal/providers/oauth2/AuthorizationActivity.java

```diff
@@ -178,11 +180,32 @@ private void extractState(final Bundle state) {
         }
     }
 
+    public static String getProcessName(@NonNull final Context context) {
+        final ActivityManager manager = (ActivityManager) context.getSystemService(Context.ACTIVITY_SERVICE);
+        for (ActivityManager.RunningAppProcessInfo processInfo : manager.getRunningAppProcesses()) {
+            if (processInfo.pid == android.os.Process.myPid()) {
+                return processInfo.processName;
+            }
+        }
+
+        return null;
+    }
+
     @Override
     protected void onCreate(final Bundle savedInstanceState) {
         final String methodName = "#onCreate";
         super.onCreate(savedInstanceState);
 
+        // Must be invoked before WebvView in common_activity_authentication is inflated.
+        // See https://developer.android.com/about/versions/pie/android-9.0-changes-28#web-data-dirs for more info.
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.P) {
+            try {
+                WebView.setDataDirectorySuffix(getProcessName(getApplicationContext()));
```

Comment:
should we check if process name is empty and only set if it's `auth` , i.e. for MSAL local auth, what would be the process name?

Reply:
fixed.